### PR TITLE
Add support for forks

### PR DESCRIPTION
**Pull request by SirSpidey**
### Description

See a comment in the internal issue \#3813 for a link to where this draft is published internally.

---
Info about contributing a new module still uses a clone. Let me know if the workflow will instead be that the contributor forks the module that we create.

---

- Adds a topic that describes how to fork a repo to update it. 
- Content that is now reused is moved to `inc`lude files.
- Changes title of topic about updating an older module to `Porting` to distinguish from new topic.

### Release required?

- [x] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: docs/update-module.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+[inc-prereqs-dev.md](inc-prereqs-dev.md ':include')
+
+## Fork, clone, and initialize the repo
+
+Fork the module repo that you want to update.
+
+1.  Navigate to the repository that you want to update.
+1.  Fork the repo by clicking **Fork** in the top-right corner of the page.
+
+    - Clear the **Copy the main branch only** option to copy all branches to your fork.
```
Comment (daniel-butler-irl) [inline docs/update-module.md:26]: I think this is optional, typically I would only want the main branch not all the forks

File: docs/update-module.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+        ```bash
+        git remote -v
+        ```
+
+    1.  If necessary, add or update the upstream repo. For example, if you use SSH, run this command:
+
+        ```bash
+        git remote add upstream git@github.com:terraform-ibm-modules/<REPO_NAME>.git
+        ```
```
Comment (daniel-butler-irl) [inline docs/update-module.md:55]: Optionally create a branch, it is not required but I would always keep work in a branch and try to keep my main fork sync'd with the upstream main.

File: docs/contribute-module.md [hunk 3]
```diff
@@ -1,27 +1,31 @@
...
 - To fix problems that are important to users.
 - To engage the community in working toward the best Terraform modules for IBM Cloud.
 
-Follow these steps to create or update a module.
+## Overall process
+
+Follow these steps to create a module.
+
+1.  Set up your development environment.
+1.  Request a new repo and clone it locally.
```
Comment (ocofaigh) [inline docs/contribute-module.md:16]: perhaps link to the request repo issue template?

File: docs/update-module.md (hunk 2 above)
Comment (SirSpidey) [reply docs/update-module.md:55]: Daniel, I've got that step in the section about [updating the terraform files](https://terraform-ibm-modules.github.io/documentation/#/contribute-module?id=update-the-terraform-files).

> From the root of the newly cloned repo, create a branch to work in by running the command git checkout -b ＜YOUR_WORKING_BRANCH>. Replace ＜YOUR_WORKING_BRANCH> with your own branch name.

**Comment by SirSpidey** [comment]
@daniel-butler-irl: Recombined the contributing topic to support both new modules and updating modules. Removed the step when forking to **Copy the main branch only**.

You can see the internal version at https://pages.github.ibm.com/allen-dean/tim-docs/#/contribute-module

**Comment by daniel-butler-irl** [comment]
Great, looks good now. One thing I saw is, we now have a doc gap. Not sure if it goes in this PR. Now we accept forks it means external contributors can contribute. These users will not have access to trigger the pipeline. The Open pull request section needs to be updated to state, if you are a member of [github-collaborators](https://github.com/orgs/terraform-ibm-modules/teams/github-collaborators) you can trigger the pipeline. Then the next part we need to figure out is for external contributors maybe ask to notify `@terraform-ibm-modules/github-admins` @ocofaigh what do you think about this?

**Review by daniel-butler-irl — APPROVED** [review]

**Review by daniel-butler-irl — DISMISSED** [review]

**Review by daniel-butler-irl — DISMISSED** [review]

**Review by daniel-butler-irl — DISMISSED** [review]

**Review by daniel-butler-irl — DISMISSED** [review]

**Review by daniel-butler-irl — APPROVED** [review]

**Review by daniel-butler-irl — APPROVED** [review]

**Review by daniel-butler-irl — APPROVED** [review]

**Review by ocofaigh — APPROVED** [review]

**Comment by SirSpidey** [comment]
Closed in favor of #52 because this PR keeps dismissing reviews.

**Pull request closed without merging**